contracts/network/contracts/Election.sol

```diff
@@ -0,0 +1,224 @@
+        Deposit[] deposits; 
+    }
+    
+    mapping (address => Validator) private validators;
+    
+    // validatorIndex contains the validator code ordered by the biggest deposit to
+    // the smallest deposit.
+    address[] validatorIndex;
+
+    // onlyWithMinDeposit requires a minimum deposit to proceed
+    modifier onlyWithMinDeposit {
+        require(msg.value >= getMinimumDeposit());
+        _;
+    }
+
+    // onlyValidator requires the sender to be a validator
+    modifier onlyValidator {
+        require(isValidator(msg.sender));
+        _;
+    }
+
+    // onlyNewCandidate required the sender to be a new candidate
+    modifier onlyNewCandidate {
+        require(!isValidator(msg.sender));
+        _;
+    }
+
+    function Election(uint _baseDeposit, uint _maxValidators, uint _unbondingPeriod, address _genesis) public {
+        require(_maxValidators >= 1);
+
+        baseDeposit = _baseDeposit;
+        maxValidators = _maxValidators;
+        unbondingPeriod = _unbondingPeriod * 1 days;
+        genesis = _genesis;
+
+        _insertValidator(_genesis, _baseDeposit);
+    }
+
+    function isGenesisValidator(address code) public view returns (bool isIndeed) {
+        return code == genesis;
+    }
+
+    function isValidator(address code) public view returns (bool isIndeed) {
+        return validators[code].isValidator;
+    }
+
+    function getValidatorCount() public view returns (uint count) {
+        return validatorIndex.length;
+    }
+
+    function getValidatorAtIndex(uint index) public view returns (address code, uint deposit) {
+        code = validatorIndex[index];
+        Validator validator = validators[code];
+        deposit = validator.deposits[validator.deposits.length - 1].amount;
+    }
+
+    function _hasAvailability() public view returns (bool available) {
+        return (maxValidators - validatorIndex.length) > 0;
+    }
+
+    // getMinimumDeposit returns the base deposit if there are positions available or
+    // the current smallest deposit required if there aren't positions availabe.
+    function getMinimumDeposit() public view returns (uint deposit) {
+        // there are positions for validator available
+        if (_hasAvailability()) {
+            return baseDeposit;
+        } else {
+            Validator smallestBidder = validators[validatorIndex[validatorIndex.length - 1]];               
+            return smallestBidder.deposits[smallestBidder.deposits.length - 1].amount + 1;
+        }
+    }
+
+    function _updateChecksum() private {
+        validatorsChecksum = keccak256(validatorIndex);
+    }
+
+    function _insertValidator(address code, uint deposit) private {
+        Validator sender = validators[code];
+        sender.index = validatorIndex.push(code) - 1;
+        sender.isValidator = true;
+        sender.deposits.push(Deposit({amount:deposit, releasedAt: 0}));
+
+        for (uint index = sender.index; index > 0; index--) {
```

Comment:
this seems to be looping for all the rest of validators and pushing the new one down in the list keeping it sorted by their latest collateral amount, right? wouldn't it be clearer to loop from `sender.index - 1` to `index == 0`? Also, just for readability, we could move this to a separate function or add a comment to this loop.